lib/dictator/not_found_handlers/default.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Dictator.NotFoundHandlers.Default do
+  @moduledoc """
+  Basic not found handler to be called if none is provided.
+
+  When a user is tries to access to a resource and none is found, a not found handler is
+  called.  This is the most basic definition.  Simply returns
+  `404 NOT FOUND` with the text "Object not found".  No
+  content type or any other header is provided.
+  """
+
+  @behaviour Plug
+
+  import Plug.Conn
+
+  @impl Plug
+  def init(_), do: :ok
+
+  @impl Plug
+  def call(conn, _) do
+    conn
+    |> send_resp(:not_found, "Object not found")
+    |> halt()
+  end
```

Comment:
expanding on my comment above, I'm not comfortable with the default response of an authorization library being a 404.

At most, this should be opt-in. Default behaviour should be the standard Unauthorized. Otherwise you risk leaking information to unauthorized users (i.e.: allowing the user to see that a certain object does not exist, even though he wouldn't have access to it in the first place)